sim/battle.ts

```diff
@@ -2820,10 +2820,10 @@ export class Battle {
 		// phazing (Roar, etc)
 		for (const side of this.sides) {
 			for (const pokemon of side.active) {
-				if (pokemon.forceSwitchFlag) {
```

Comment:
The fact that TypeScript isn't complaining about this implies that null pokemon shouldn't exist here. So... my best guess is that somehow this is getting run after a battle's been destroyed? And so that's the bug?

Or maybe some OM is having null pokemon, and that's the bug?